tensorflow/tsl/platform/cpu_info.cc

```diff
@@ -414,10 +417,22 @@ class CPUIDInfo {
       if (static_cast<bool>(getline(midr_el1_file, line))) {
         uint32 midr_el1 = std::stoul(line, nullptr, 16);
 
-        // Unpack variant and CPU ID.
+        // Unpack variant and CPU ID
+        // Reference:
+        // https://developer.arm.com/documentation/101427/0101/Register-descriptions/AArch64-system-registers/MIDR-EL1--Main-ID-Register--EL1.
         cpuid->implementer_ = (midr_el1 >> 24) & 0xFF;
         cpuid->variant_ = (midr_el1 >> 20) & 0xF;
         cpuid->cpunum_ = (midr_el1 >> 4) & 0xFFF;
+        if (cpuid->implementer_ == 0x41) {
+          switch (cpuid->cpunum_) {
+            case 0xd40:  // ARM NEOVERSE V1
+              cpuid->is_arm_neoverse_v1_ = 1;
```

Comment:
note: this falls through to the next statement, leading to compile warnings which are converted to errors on our end.  I bet penporn@ is going to fix this internally as well ;).  But in future, if this was intentional, we need to annotate with `ABSL_FALLTHROUGH_INTENDED` (or just avoid fall-throughs).

Reply:
Nice catch! :)
In the previous version of the code that I reviewed, each case just returns a value, e.g., 
```
case ARM_NEOVERSE_V1: return cpuid->is_arm_neoverse_v1_;
``` 
[like here](https://github.com/tensorflow/tensorflow/pull/60353#discussion_r1275374092) (the commit was overwritten and can't be seen on PR #60353 anymore). These changes were moved from #60353 so I didn't see that there were changes to this part of code. 

I don't think a fallthrough is intended. I'll add `break`s internally.

Reply:
Thanks @cantonios and @penpornk. Yes it isn't supposed to fall through. Thanks for fixing @penpornk.

Reply:
Thanks for the fix